src/utils/resampling.ts

```diff
@@ -2,47 +2,46 @@
  * Resizes a given array to consist of more elements than provided. This uses
  * linear interpolation to fill in the gaps.
  *
- * @param data {number[]} the list of data points to interpolate
- * @param threshold {number} the number of expected data points from the array
- * @returns {number[]} the array with the required number of points
+ * @param data the list of data points to interpolate
+ * @param threshold the number of expected data points from the array
+ * @returns the array with the required number of points
  */
-export let upsampleArray = (data, threshold) => {
-  let linearInterpolate = function (before, after, atPoint) {
-    return before + (after - before) * atPoint
-  }
+export const upsampleArray = (data: number[], threshold: number): number[] => {
+  const linearInterpolate = (before: number, after: number, atPoint: number) =>
+    before + (after - before) * atPoint
 
-  let newData = []
-  let springFactor = (data.length - 1) / (threshold - 1)
+  const newData = []
+  const springFactor = (data.length - 1) / (threshold - 1)
   newData[0] = data[0] // for new allocation
   for (let i = 1; i < threshold - 1; i++) {
-    let tmp = i * springFactor
-    let before = Math.floor(tmp).toFixed()
-    let after = Math.ceil(tmp).toFixed()
-    let atPoint = tmp - before
+    const tmp = i * springFactor
+    const before = Math.floor(tmp)
+    const after = Math.ceil(tmp)
+    const atPoint = tmp - before
     newData[i] = linearInterpolate(data[before], data[after], atPoint)
   }
   newData[threshold - 1] = data[data.length - 1] // for new allocation
   return newData
 }
 
 /**
- * Resizes a given array to consist of less elements than provided. This uses
+ * Resizes a given array to consist of fewer elements than provided. This uses
  * the Largest Triangle Three Buckets algorithm by Sveinn Steinarsson.
  *
  * @see {@link https://github.com/sveinn-steinarsson/flot-downsample}
  *
- * @param data {number[]} the list of data points to interpolate
- * @param threshold {number} the number of expected data points from the array
- * @returns {number[]} the array with the required number of points
+ * @param data - the list of data points to interpolate
+ * @param threshold - the number of expected data points from the array
```

Comment:
Added this in 6a3b521. Have currently restricted this rule to only apply to `.ts` files.

Reply:
If you don't mind, it'd probably be better to add it to #999 or just put it into a separate PR entirely. I'd like to go ahead and apply the rules to JS, TS and Vue SFCs as well in one go.

Reply:
Adding this to JS and Vue is leading to a lot of errors because of the `typedef` comments. It might be good to re-enable it for JS after a major chunk of the migration is complete.

Reply:
Yeah, I think it'd be better to use a generic JSDoc one than a TSDoc thing as TSDoc is actually a specific _separate_ standard than JSDoc with different rules in some cases.